lms/views/lti/basic_launch.py

```diff
@@ -291,15 +278,20 @@ def _do_launch(self, document_url, grading_supported=True):
         if grading_supported:
             self.context.js_config.maybe_enable_grading()
 
+        self.context.js_config.enable_lti_launch_mode()
+        self.context.js_config.maybe_set_focused_user()
         self.context.js_config.add_document_url(document_url)
 
         return {}
 
     def _store_lti_data(self):
         """Store LTI launch data in our LMS database."""
-
         request = self.request
 
+        request.find_service(name="application_instance").get_current().update_lms_data(
```

Comment:
This _could_ make sense in the constructor.

The main thing update_lms_data is doing is potentially raising `ReusedConsumerKey`, yeah it might update some data like it says in the tin but the raising part is what matters most.